### PR TITLE
WFLY-9325 - improve execution time of mixed-domain testsuite

### PR DESCRIPTION
**Pull request by ctomc**
- remove unneded bin/ and docs/ folder copying
- remove unnecessary deps in mixed domain testsuite

In some quick tests done on CI, it shows that this changes do bring speed improvement to mixed-domain testsuite (which was main focus of this PR)
- about 20% on linux machines (tested on CI) 04:37 min vs 05:07 min
- about 30% on windows machines (CI) 06:25 min vs 07:45 min

improvements are also in other testsuites, as we stop copying of bin/client folder that has 25mb, but main focus was mixed-domain here.

### CODE REVIEW

File: testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/ExpressionSupportSmokeTestCase.java [hunk 1]
```diff
@@ -157,6 +157,7 @@ public void test() throws Exception {
     public void setUp() throws IOException {
         final WildFlyManagedConfiguration config = createConfiguration("domain.xml", "host.xml", getClass().getSimpleName());
         config.setAdminOnly(true);
+        //  config.setReadOnlyDomain(true);
```
Comment (bstansberry) [inline testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/ExpressionSupportSmokeTestCase.java]: Can you remove that please?

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/web/headers/ResponseCodeTestCase.java [hunk 2]
```diff
@@ -110,7 +110,8 @@ public void testServerInfo() throws Exception {
          final HttpEntity entity = response.getEntity();
          Assert.assertNotNull("Null entity!", entity);
          final String content = EntityUtils.toString(response.getEntity());
-         Assert.assertTrue("Wrong content! " + content, content.matches("WildFly Full .*\\(WildFly Core .*\\) - .*"));
+         //Assert.assertTrue("Wrong content! " + content, content.matches("WildFly Full .*\\(WildFly Core .*\\) - .*"));
+         Assert.assertTrue("Wrong content! " + content, content.contains("WildFly"));
```
Comment (bstansberry) [inline testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/web/headers/ResponseCodeTestCase.java]: Why this change?

File: testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/BuildConfigurationTestBase.java [hunk 3]
```diff
@@ -144,35 +143,34 @@ private static File hackFixHostConfig(File hostConfigFile, String hostName, Stri
...
-                    }
-                    writer.write("\n");
-                    line = reader.readLine();
+        try (BufferedReader reader = new BufferedReader(new FileReader(domainConfigFile));
+             final BufferedWriter writer = Files.newBufferedWriter(file.toPath())
+        ) {
+            String line = reader.readLine();
+            while (line != null) {
+                if (line.contains("<security-setting name=\"#\">")) { //super duper hackish, just IO optimization
+                    writer.write("        <journal file-size=\"1024\" />");
```
Comment (bstansberry) [inline testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/BuildConfigurationTestBase.java]: Ugh. But ok I guess.

**Review by bstansberry — CHANGES_REQUESTED** [review]
Couple small points aside, looks good. Anything noteworthy in org.jboss:jboss-parent:25?

File: testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/BuildConfigurationTestBase.java (hunk 3 above)
Comment (ctomc) [reply testsuite/domain/src/test/java/org/jboss/as/test/integration/domain/BuildConfigurationTestBase.java]: without this change, we by default on boot create 24mb of Artemis journal files. 
with this change this goes down to ~2mb

File: testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/web/headers/ResponseCodeTestCase.java (hunk 2 above)
Comment (ctomc) [reply testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/web/headers/ResponseCodeTestCase.java]: long story, will remove it for now.

**Comment by ctomc** [comment]
@bstansberry probably the most noteworthy change in jboss-parent is newer version of enforcer-plugin which is *much* faster when old one.
+ others are related to plugin updates that help with JDK9

**Review by bstansberry — APPROVED** [review]
